### PR TITLE
Mark kins in KS test result output

### PR DESCRIPTION
**Pull request by the-x-at**
A new column `ksgrp` is added to the Kinship sum test, which groups affected individuals by kinship. Solves issue #15.

- [x]  `R CMD check` runs successfully.

### CODE REVIEW

File: R/Methods-FAKinSumResults.R [hunk 1]
```diff
@@ -147,16 +149,45 @@ setMethod("result", "FAKinSumResults", function(object, method="BH"){
...
-
+##' Define groups of related affected individuals for KS test.
+##' @param kin Kinship matrix.
+##' @param padj Named vector with adjusted P values of all affected individuals.
+##' @param th  Threshold of `padj` for inclusion of affected individuals.
+##' @return A named vector of group numbers or NAs, the names correspond to the
+##' IDs of the affected individuals. An NA indicates that the affected
+##' individual was above the threshold. The named vector corresponds to the
+##' entries in vector `padj`.
+define.result.groups.ks <- function(kin, padj, th=0.05)
```
Comment (jorainer) [inline R/Methods-FAKinSumResults.R]: should we maybe also add an additional parameter `minKinship = 0` to the function (and then also to `doShareKinship`) that allows to define which individuals to group? I guess we would not like to put e.g. 3rd cousins into the same "kin" group as their kin is only very low. The default could be still `0` but we could then also use a different kinship threshold - as I would like to not put, like said, third grade cousins into the same kinship group.

**Review by jorainer — CHANGES_REQUESTED** [review]
This is an extremely helpfull additional functionality @the-x-at ! I have just one comment we should discuss before merging.

File: R/Methods-FAKinSumResults.R (hunk 1 above)
Comment (the-x-at) [reply R/Methods-FAKinSumResults.R]: Sure, that's easy.

Comment (jorainer) [reply R/Methods-FAKinSumResults.R]: Excellent! Please re-request my review once you've pushed to this PR and are OK with me reviewing it.

**Comment by the-x-at** [comment]
Was more work than I assumed. `shareKinship` exists in multiple locations. Documentation had to be updated. `R-4.1.0` was just released with Bioconductor 3.13, also tested it there. Overall, see my commit 7f2a396 for a full list of changes. I decided to call the new parameter `rmKinship`, as `minKinship` was misleading: it removes all kins with kinship ≤ `rmKinship`, whereas `minKinship` would indicate that the value itself would still be included.

**Review by jorainer — APPROVED** [review]
Looks good to me! Will fix the merge conflicts and CI and then merge.

**Pull request merged**